src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -111,17 +111,30 @@ public override int GetHashCode()
 
         public override ImmutableArray<DiagnosticAnalyzer> GetAnalyzersForAllLanguages()
         {
-            return _diagnosticAnalyzers.GetExtensionsForAllLanguages();
+            // This API returns duplicates of analyzers that support multiple languages.
+            // We explicitly retain this behaviour to ensure back compat
+            return _diagnosticAnalyzers.GetExtensionsForAllLanguages(includeDuplicates: true);
         }
 
         public override ImmutableArray<DiagnosticAnalyzer> GetAnalyzers(string language)
         {
             return _diagnosticAnalyzers.GetExtensions(language);
         }
 
+        public override ImmutableArray<ISourceGenerator> GetGeneratorsForAllLanguages()
+        {
+            return _generators.GetExtensionsForAllLanguages(includeDuplicates: false);
+        }
+
+        [Obsolete("Use GetGenerators(string language) or GetGeneratorsForAllLanguages()")]
         public override ImmutableArray<ISourceGenerator> GetGenerators()
```

Comment:
📝@333fred I'd like to run this by the API compat review board.

Reply:
In a perfect world, I'd just remove it. Right now I've obsoleted it, and it returns only CSharp generators, which is essentially what it did before.